synapse/handlers/presence.py

```diff
@@ -951,11 +981,13 @@ async def _handle_timeouts(self) -> None:
 
         syncing_user_ids = {
             user_id
-            for user_id, count in self.user_to_num_current_syncs.items()
+            for (user_id, _), count in self._user_device_to_num_current_syncs.items()
             if count
         }
-        for user_ids in self.external_process_to_current_syncs.values():
-            syncing_user_ids.update(user_ids)
+        for user_id, _ in itertools.chain(
+            *self.external_process_to_current_syncs.values()
+        ):
+            syncing_user_ids.add(user_id)
```

Comment:
I think this would likely be better as an `.update(..)`?

```suggestion
        syncing_user_ids.update(
            user_id
            for user_id, _ in itertools.chain(
                *self.external_process_to_current_syncs.values()
            )
        )
```

Rather than adding individually? Not actually sure if it makes a difference in practice

Reply:
I do this in the next PR and failed to update this one. Thanks! 👍